mex.go

```diff
@@ -326,3 +340,23 @@ func (mexset *messageExchangeSet) forwardPeerFrame(frame *Frame) error {
 
 	return nil
 }
+
+// stopExchanges goes through all the exhanges and makes sure nobody is waiting
+// because the connection is already being run down
+func (mexset *messageExchangeSet) stopExchanges(err error) error {
+	if mexset.log.Enabled(LogLevelDebug) {
+		mexset.log.Debugf("stopping exchangeset %s", mexset.name)
+	}
+
+	// We need to stop all exchanges on this set
+	mexset.RLock()
+	if mexset.log.Enabled(LogLevelDebug) {
+		mexset.log.Debugf("stopping num exchanges: %v", len(mexset.exchanges))
```

Comment:
Avoid 2 logs in the same method, have a single log, and it should be something along the lines of:

```go
mexset.log.Debugf("stopping %v exchanges due to error: %v", numExchanges, err)
```
the name of the mexset should be part of `mexset.log` as a field so you don't need to duplicate it